File: bookstore-quarkus/src/main/kotlin/com/arconsis/http/books/Book.kt
package com.arconsis.http.books

import com.arconsis.common.UUIDSerializer
import kotlinx.datetime.LocalDate
import kotlinx.serialization.Serializable
import java.util.*

@Serializable
data class Book(
    @Serializable(with = UUIDSerializer::class)
    val id: UUID,
    val title: String,
    val author: String,
    val publisher: String,
    val releaseDate: LocalDate
)

File: bookstore-rocketrs/src/models/book.rs
use rocket::serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Book {
    #[serde(default)]
    pub id: String,
    pub title: String,
    pub author: String,
    #[serde(rename = "releaseDate")]
    pub release_date: String,
    pub publisher: String,
}
